testapp/testapp/AppDelegate.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  AppDelegate.swift
+//  testapp
+//
+//  Created by Christopher Batin on 16/01/2019.
+//  Copyright © 2019 Christopher Batin. All rights reserved.
+//
+
+import UIKit
+import Ably
+import UserNotifications
+
+// 1
+let authURL = "YOUR_NGROK_HTTPS/auth"
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate, ARTPushRegistererDelegate {
+    var window: UIWindow?
+    var realtime: ARTRealtime!
+    var channel: ARTRealtimeChannel!
+    var subscribed = false
+
+    //2
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        DispatchQueue.main.async() {
```

Comment:
@cjbatin Why do we need an async call? Ant particular reason?

Reply:
Without the async I wasn't receiving anything. Better to be safe than sorry anyway

Reply:
@cjbatin Sorry, if it's not working without the dispatch then something is wrong. We should not ignore what's suppose to work. This tutorial should be straight and focused, no need to add more boilerplate that could confuse the user library. /cc @Srushtika